### PR TITLE
Export templates using ssh

### PR DESCRIPTION
**Pull request by peterdragun**
Adding test for exporting templates to git using ssh. Also adding test to cover BZ: [1785613](https://bugzilla.redhat.com/show_bug.cgi?id=1785613)
~~To reduce number of similar test I used test parametrization. Using different fixtures in one test required one additional package `pytest_lazy-fixture` which basically takes name of fixture and evaluates it.~~ Updated to use indirect parametrization.

Test results:
```
pytest tests/foreman/*/test_templatesync.py
================================================================ test session starts ================================================================
platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo, configfile: pyproject.toml
plugins: cov-2.12.1, forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, ibutsu-1.16, xdist-2.4.0, lazy-fixture-0.6.3
collected 43 items                                                                                                                                  

tests/foreman/api/test_templatesync.py .............................                                                                          [ 67%]
tests/foreman/cli/test_templatesync.py ........                                                                                               [ 86%]
tests/foreman/ui/test_templatesync.py ......                                                                                                  [100%]

==================================================== 43 passed, 0 warnings in 485.91s (0:08:05) ====================================================
```

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Can one of the admins verify this patch?

**Review by ogajduse — APPROVED** [review]
This is a great addition to the Template Sync component!  The component would be well covered with it!
I do not have any objections to using additional dependency, it saves us a lot of trouble here.

I would like to ask @mshriver and @JacobCallahan for a review.

File: tests/foreman/api/test_templatesync.py [hunk 1]
```diff
@@ -323,7 +330,7 @@ def test_positive_import_and_associate(
         assert len(ptemplate[0].read().organization) == 1
 
     @pytest.mark.tier2
-    def test_positive_import_to_subdirectory(self, module_org):
+    def test_positive_import_from_subdirectory(self, module_org):
```
[the remark this answers is not in the document]
Comment (mshriver) [reply tests/foreman/api/test_templatesync.py:333]: Yep, we match by UUID

File: tests/foreman/api/test_templatesync.py [hunk 2]
```diff
@@ -1056,3 +1021,187 @@ def test_positive_export_log_to_production(
...
+        [
+            f'http://{git.username}:{git.password}@{git.hostname}:{git.http_port}',
+            f'ssh://git@{git.hostname}:{git.ssh_port}',
+        ],
+        ids=['http', 'ssh'],
+    )
+    @pytest.mark.parametrize(
+        'repo, branch',
+        [
+            (lazy_fixture(('git_repository', 'git_branch'))),
```
Comment (mshriver) [inline tests/foreman/api/test_templatesync.py:1038]: I don't think this dependency is strictly required for what you're trying to do here.

Whether the created repository has `init=True` or `init=False` within the fixture could be indirectly parametrized.
The branch can also be indirectly parametrized.

Comment (peterdragun) [reply tests/foreman/api/test_templatesync.py:1038]: But this would mean that every time test is created it will also create new repository or not?

Comment (mshriver) [reply tests/foreman/api/test_templatesync.py:1038]: The scope of the fixtures still applies when indirectly parametrizing, in this case you have session scoped fixtures (I'd recommend module scope since they're defined in the local test module).

**Comment by JacobCallahan** [comment]
@peterdragun can you elaborate on what role `lazy_fixture` fills here and why normal fixtures won't work? I'm not seeing a reason for the new dependency at the moment, so clarification would be appreciated.

**Comment by peterdragun** [comment]
@JacobCallahan 
As I mentioned in comment:
The main reason why I wanted to use lazy_fixture is that it allows me to use different scopes for different fixtures. It doesn't make sense to create repository for every single test, this is reason why I set scope to session. But on the other hand it doesn't make sense to use empty repository in more than one test, because it won't be empty anymore.

But, I mean, all fixtures could have function scope but it will take a lot more api calls to gitea, considering how long it takes for one request it is not really a big issue now. And I can use indirect parametrization if this is better option for you.

**Comment by peterdragun** [comment]
Updated to use indirect parametrization of fixture.
New test results:
```
pytest -v tests/foreman/*/test_templatesync.py
======================================================================= test session starts ========================================================================
platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1 -- /home/pdragun/.virtualenvs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo, configfile: pyproject.toml
plugins: cov-2.12.1, forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, ibutsu-1.16, xdist-2.4.0
collected 43 items                                                                                                                                                 

tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_filtered_templates_from_git PASSED                                    [  2%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_import_filtered_templates_from_git_with_negate PASSED                                 [  4%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_and_associate PASSED                                                  [  6%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_from_subdirectory PASSED                                              [  9%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_localdir PASSED                                 [ 11%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_negate PASSED                                      [ 13%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_and_import_with_metadata PASSED                                       [ 16%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_verbose[True] PASSED                                      [ 18%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_verbose[False] PASSED                                     [ 20%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_changed_key_true PASSED                                   [ 23%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_changed_key_false PASSED                                  [ 25%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_name_key PASSED                                           [ 27%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_imported_key PASSED                                       [ 30%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_file_key PASSED                                           [ 32%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_corrupted_metadata PASSED                                 [ 34%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_filtered_skip_message PASSED                              [ 37%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_no_name_error PASSED                                      [ 39%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_no_model_error PASSED                                     [ 41%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_json_output_blank_model_error PASSED                                  [ 44%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_json_output PASSED                                                    [ 46%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_log_to_production PASSED                                              [ 48%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_log_to_production PASSED                                              [ 51%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_all_templates_to_repo[non_empty_repo-http] PASSED                     [ 53%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_all_templates_to_repo[non_empty_repo-ssh] PASSED                      [ 55%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_all_templates_to_repo[empty_repo-http] PASSED                         [ 58%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_all_templates_to_repo[empty_repo-ssh] PASSED                          [ 60%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_all_templates_from_repo PASSED                                        [ 62%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_negative_import_locked_template PASSED                                                [ 65%]
tests/foreman/api/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_locked_template PASSED                                                [ 67%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_import_force_locked_template PASSED                                          [ 69%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_update_templates_in_git[non_empty_repo-http] PASSED                          [ 72%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_update_templates_in_git[non_empty_repo-ssh] PASSED                           [ 74%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_git[non_empty_repo-http] PASSED                 [ 76%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_git[non_empty_repo-ssh] PASSED                  [ 79%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_git[empty_repo-http] PASSED                     [ 81%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_git[empty_repo-ssh] PASSED                      [ 83%]
tests/foreman/cli/test_templatesync.py::TestTemplateSyncTestCase::test_positive_export_filtered_templates_to_temp_dir PASSED                                 [ 86%]
tests/foreman/ui/test_templatesync.py::test_positive_import_templates PASSED                                                                                 [ 88%]
tests/foreman/ui/test_templatesync.py::test_positive_export_templates PASSED                                                                                 [ 90%]
tests/foreman/ui/test_templatesync.py::test_positive_export_filtered_templates_to_git[non_empty_repo-http] PASSED                                            [ 93%]
tests/foreman/ui/test_templatesync.py::test_positive_export_filtered_templates_to_git[non_empty_repo-ssh] PASSED                                             [ 95%]
tests/foreman/ui/test_templatesync.py::test_positive_export_filtered_templates_to_git[empty_repo-http] PASSED                                                [ 97%]
tests/foreman/ui/test_templatesync.py::test_positive_export_filtered_templates_to_git[empty_repo-ssh] PASSED                                                 [100%]

=========================================================== 43 passed, 75 warnings in 477.53s (0:07:57) ============================================================
```

**Review by ogajduse — APPROVED** [review]

File: pytest_fixtures/component/templatesync.py [hunk 3]
```diff
@@ -36,29 +34,56 @@ def create_import_export_local_dir(default_sat):
...
-    port = urlparse(settings.git.url).port
-    if port:
-        default_sat.execute(f'semanage port -a -t http_port_t -p tcp {port}')
+    default_sat.execute(f'semanage port -a -t http_port_t -p tcp {settings.git.http_port}')
+    default_sat.execute(f'semanage port -a -t ssh_port_t -p tcp {settings.git.ssh_port}')
 
 
 @pytest.fixture(scope='session')
-def git_repository(git_port):
+def git_pub_key(default_sat, git_port):
```
Comment (jyejare) [inline pytest_fixtures/component/templatesync.py:42]: This is a good `core` fixture, move this to `core/sys.py` !

Comment (ogajduse) [reply pytest_fixtures/component/templatesync.py:42]: @jyejare Do you see any other current or future tests that might use this fixture?
This fixture is tied to Gitea/Github service. I see no other place in Satellite where git is used. Do you see any?

**Comment by JacobCallahan** [comment]
@peterdragun thanks for the change!

**Review by swadeley — APPROVED** [review]
ACK
